src/java.base/share/classes/sun/net/www/protocol/ftp/Handler.java

```diff
@@ -58,6 +59,12 @@ protected java.net.URLConnection openConnection(URL u)
 
     protected java.net.URLConnection openConnection(URL u, Proxy p)
         throws IOException {
-        return new FtpURLConnection(u, p);
+        FtpURLConnection connection = null;
+        try {
+            connection = new FtpURLConnection(u, p);
+        } catch (IllegalArgumentException e) {
+            throw new MalformedURLException(e.getMessage());
```

Comment:
I'd suggest keeping the IllegalArgumentException as the root cause of the MUE for better diagnosability. Otherwise LGTM.

Reply:
The constructor "MalformedURLException (Throwable cause)" doens'nt exist. Isn't there any other method?

Reply:
You can use `Throwable::initCause`

Reply:
```
var mfue = new MalformedURLException(e.getMessage());
mfue.initCause(e);
throw mfue;
```

Reply:
Thank you for your advice, I could fix it.